pkg/compiler/visitor.go

```diff
@@ -50,6 +55,69 @@ func (v *visitor) VisitProgram(ctx *fql.ProgramContext) interface{} {
 	})
 }
 
+func (v *visitor) doVisitHeads(heads []fql.IHeadContext) error {
+	for _, head := range heads {
+		err := v.doVisitHead(head.(*fql.HeadContext))
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (v *visitor) doVisitHead(head *fql.HeadContext) error {
+	useexpr := head.UseExpression().(*fql.UseExpressionContext)
+
+	// TODO: Think about improving collision analysis to display more detailed errors.
+	// For example, "mamespaces X and Y both contain function F"
+	if iuse := useexpr.Use(); iuse != nil {
+		ns := iuse.(*fql.UseContext).
+			NamespaceIdentifier().
+			GetText()
+
+		err := deleteNamespace(v.funcs, ns)
```

Comment:
why are you removing it?
Can't I use something like this:
```
USE IO::FS

LET content = READ(@file)

IO::FS::WRITE(@anotherFile, content)

RETURN NONE
```

Reply:
I don’t see cases where this could be useful.  Maybe you have some examples?


I think this is a more clean approach. Always use with the prefix, or without.

Reply:
It might be not very intuitive for people coming from another languages where both cases would work (Java, C#, Rust)

Reply:
Here is a use case. I'm working on an assertion library for Ferret and there are 2 functions that may cause some problems: ``T::TRUE`` and ``T::FALSE``.
If you don't allow to use full path, there will be a collision with global boolean values ``TRUE`` and ``FALSE``.